core/trino-main/src/main/java/io/trino/cost/JoinStatsRule.java

```diff
@@ -180,47 +182,61 @@ private PlanNodeStatsEstimate filterByEquiJoinClauses(
+            return PlanNodeStatsEstimate.unknown();
         }
 
-        return result;
-    }
+        PlanNodeStatsEstimate combinedEstimate = knownSortedEstimates.poll().getEstimate();
+        double combinedSelectivity = combinedEstimate.getOutputRowCount() / stats.getOutputRowCount();
+        double combinedIndependenceFactor = 1.0;
+        double independenceFactor = getJoinMultiClauseIndependenceFactor(session);
+        // For independenceFactor = 0.75 and terms t1, t2, t3
+        // Combined selectivity = (t1 selectivity) * ((t2 selectivity) ^ 0.75) * ((t3 selectivity) ^ (0.75 * 0.75))
+        // independenceFactor = 1 implies the terms are assumed to have no correlation and their selectivities are multiplied without scaling.
+        // independenceFactor = 0 implies the terms are assumed to be fully correlated and only the most selective term drives the selectivity.
+        while (!knownSortedEstimates.isEmpty()) {
+            PlanNodeStatsEstimateWithClause estimateWithClause = knownSortedEstimates.poll();
+            PlanNodeStatsEstimate clauseEstimate = estimateWithClause.getEstimate();
+            EquiJoinClause clause = estimateWithClause.getClause();
+            combinedIndependenceFactor *= independenceFactor;
+            double scaledClauseSelectivity = Math.pow(clauseEstimate.getOutputRowCount() / stats.getOutputRowCount(), combinedIndependenceFactor);
+            // If scaledClauseSelectivity goes above UNKNOWN_FILTER_COEFFICIENT threshold,
+            // fall back to multiplying output row count by UNKNOWN_FILTER_COEFFICIENT
+            if (scaledClauseSelectivity >= UNKNOWN_FILTER_COEFFICIENT) {
+                knownSortedEstimates.add(estimateWithClause);
+                break;
+            }
+            combinedSelectivity *= scaledClauseSelectivity;
+            double outputRowCount = stats.getOutputRowCount() * combinedSelectivity;
+            combinedEstimate = filterByClause(combinedEstimate, clause, outputRowCount, types);
+        }
 
-    private PlanNodeStatsEstimate filterByEquiJoinClauses(
-            PlanNodeStatsEstimate stats,
-            EquiJoinClause drivingClause,
-            Collection<EquiJoinClause> remainingClauses,
-            Session session,
-            TypeProvider types)
-    {
-        ComparisonExpression drivingPredicate = new ComparisonExpression(EQUAL, drivingClause.getLeft().toSymbolReference(), drivingClause.getRight().toSymbolReference());
-        PlanNodeStatsEstimate filteredStats = filterStatsCalculator.filterStats(stats, drivingPredicate, session, types);
-        for (EquiJoinClause clause : remainingClauses) {
-            filteredStats = filterByAuxiliaryClause(filteredStats, clause, types);
+        while (!knownSortedEstimates.isEmpty()) {
+            EquiJoinClause clause = knownSortedEstimates.poll().getClause();
+            double outputRowCount = combinedEstimate.getOutputRowCount() * UNKNOWN_FILTER_COEFFICIENT;
```

Comment:
In filter we use `hasUnestimatedTerm`. However, here we multiply by `UNKNOWN_FILTER_COEFFICIENT` for every term. Could we unify that one way or the other?

Reply:
For the time being I'm going to stick to existing behaviour around unestimated terms, we can revisit it another PR